open_instruct/ground_truth_utils.py

```diff
@@ -635,14 +638,60 @@ async def async_call(
         retry_delay = 1.0
 
         for attempt in range(max_retries):
+            # judges the quality of a response
             try:
-                messages = build_messages(prompt)
+                system_prompt = "Do not generate text between the <think> and </think> tags." # "You are a concise assistant who gives very short explanations before giving a quality score."
+                messages = build_messages(prompt, system_prompt)
+
+                # Faeze: check if the request would exceed context window
+                # Import the context window checker
+                try:
+                    from open_instruct.context_window_checker import (
+                        check_context_window_limit,
+                        truncate_messages_to_fit_context,
+                    )
+
+                    context_check_available = True
+                except ImportError:
+                    logger.warning("Context window checker not available. Proceeding without context checking.")
+                    context_check_available = False
+
+                # Check if the request would exceed context window
+                if context_check_available:
```

Comment:
should we also check if we have a compatible tokenizer? For example, I don't think litellm supports the olmo tokenizer.

Reply:
sounds good!